CMakeLists.txt

```diff
@@ -76,6 +82,15 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS
     CACHE STRING "Enable/Disable output of compile commands during generation." FORCE
 )
 
+# Enable all warnings that we can
+add_compile_options(-Wall -Wpedantic -Wextra)
+
+# This is needed everywhere so that we can catch segmentation faults
+add_compile_options(-fnon-call-exceptions)
+
+# Make NUClear use the system clock, as we are syncing it with ptp
+add_compile_definitions(NUCLEAR_CLOCK_TYPE=std::chrono::steady_clock)
```

Comment:
I presume this plays nicely with and without a custom clock?

Reply:
With the latest version of NUClear it does, I should check you guys are using that